server/job/blacklistedTagsProcessor.ts

```diff
@@ -0,0 +1,186 @@
+          label: 'Jobs',
+        });
+      } else {
+        throw err;
+      }
+    } finally {
+      this.reset();
+    }
+  }
+
+  public status(): StatusBase {
+    return {
+      running: this.running,
+      progress: this.progress,
+      total: this.total,
+    };
+  }
+
+  public cancel() {
+    this.running = false;
+    this.progress = 0;
+    this.total = 0;
+  }
+
+  private reset() {
+    this.cancel();
+  }
+
+  private async createBlacklistEntries(em: EntityManager) {
+    const tmdb = createTmdbWithRegionLanguage();
+
+    const settings = getSettings();
+    const blacklistedTags = settings.main.blacklistedTags;
+    const blacklistedTagsArr = blacklistedTags.split(',');
+
+    const pageLimit = settings.main.blacklistedTagsLimit;
+
+    if (blacklistedTags.length === 0) {
+      return;
+    }
+
+    // The maximum number of queries we're expected to execute
+    this.total =
+      2 * blacklistedTagsArr.length * pageLimit * SortOptionsIterable.length;
+
+    for (const type of [MediaType.MOVIE, MediaType.TV]) {
+      const getDiscover =
+        type == MediaType.MOVIE ? tmdb.getDiscoverMovies : tmdb.getDiscoverTv;
+
+      // Iterate for each tag
+      for (const tag of blacklistedTagsArr) {
+        let queryMax = pageLimit * SortOptionsIterable.length;
+        let fixedSortMode = false; // Set to true when the page limit allows for getting every page of tag
+
+        for (let query = 0; query < queryMax; query++) {
+          const page: number = fixedSortMode
+            ? query + 1
+            : (query % pageLimit) + 1;
+          const sortBy: SortOptions | undefined = fixedSortMode
+            ? undefined
+            : SortOptionsIterable[query % SortOptionsIterable.length];
+
+          if (!this.running) {
+            throw new AbortTransaction();
+          }
+
+          const response = await getDiscover({
+            page,
+            sortBy,
+            keywords: tag,
+          });
+          await this.processResults(response, tag, type, em);
+          await new Promise((res) => setTimeout(res, TMDB_API_DELAY_MS));
+
+          this.progress++;
+          if (page == 1 && response.total_pages <= queryMax) {
+            // We will finish the tag with less queries than expected, move progress accordingly
+            this.progress += queryMax - response.total_pages;
+            fixedSortMode = true;
+            queryMax = response.total_pages;
+          }
+        }
+      }
+    }
+  }
+
+  private async processResults(
+    response: TmdbSearchMovieResponse | TmdbSearchTvResponse,
+    keywordId: string,
+    mediaType: MediaType,
+    em: EntityManager
+  ) {
+    const blacklistRepository = em.getRepository(Blacklist);
+
+    for (const entry of response.results) {
+      const blacklistEntry = await blacklistRepository.findOne({
+        where: { tmdbId: entry.id },
+      });
+
+      if (blacklistEntry != null) {
+        // Don't mark manual blacklists with tags
```

Comment:
Why?

Reply:
Because they weren't blacklisted for a tag, they were blacklisted manually. If I added a tag to the db, it would appear in the blacklist as blacklisted by a tag rather than a person. It would also mean it would get dropped at the start of the job, but not re-added if the job didn't find it in one of the tags. It could also get removed if a tag is removed from the setting.

Since it was blacklisted by a human, I felt it should remain as it did before and the job shouldn't touch it.